### DISCUSSION TITLE
SQS based scaling behavior?

### DISCUSSION THREAD

**Discussion by venkat-raman-wday** [kedacore/keda#4439] [Q&A / Need Help]
Hello,

We've setup Keda to scale our pods based on SQS.

Following is our Keda setup for SQS based scaling,

- set ```queueLength``` to 1, since our model follows a one message per pod.
- set ```pollInterval``` to 1, since we'd like to scale immediately.
- set ```maxReplicaCount``` to 5 since we run this setup in a small cluster.

We only care about messages that are not in-flight. What's happening is, if there are 6 messages in the queue, there are 5 pods spun up as expected since the ```maxReplicaCount``` is set to 5. However, when one of them completes/finishes their work, a new pod is not being spun up to scale for the pending message.
The expected behavior would be that the pending message be scheduled immediately to a scaled pod while not going above the set replica count.

Sometimes, scaling doesn't happen if the requests aren't filling the queue fast enough starting with an empty queue, despite the number of pods not reaching the ```maxReplicaCount```.

Please correct me if my understanding is incorrect or what might be wrong?

**Comment by JorTurFer** [comment]
hi,
Just to understand your use case. You are using ScaledObject and you have some pods consuming messages, right? I don't get this part:
> However, when one of them completes/finishes their work, a new pod is not being spun up to scale for the pending message.

Do your pods process only one message per pod and then finish? I mean, if your pods are workers consuming messages constantly, you don't need to spin up other pod because all of them are still alive.

I guess that you use ScaledJobs for it (because it matches better with your use case). If this is your case, I think that the problem is related with not taking into account `in-flight` messages, I mean, if you have 6 messages and 0 pods, KEDA will spin up 5 pods (maxReplicaCount) and they will take one message per pod, so you will have 5 in-flight and 1 pending. When a pod finishes, you have 4 in-fight and 1 pending, but you still have 4 jobs in progress, so KEDA thinks that no more pods are needed. 
I'd suggest trying with in-flight enabled because the case would be, 4 in-flight + 1 pending = 5 items but only 4 jobs => KEDA will spin up another job
